plop-templates/@react-spectrum/src/Component.tsx.hbs

```diff
@@ -10,33 +10,65 @@
  * governing permissions and limitations under the License.
  */
 
-import {classNames, DOMRef, filterDOMProps, useDOMRef, useStyleProps} from '@react-spectrum/utils';
-import {mergeProps} from '@react-aria/utils';
-import { {{~componentName}}Props} from '@react-types/${{packageName}}';
+import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+{{#if (includes scopes "@react-types")}}
+import {DOMRef} from '@react-types/shared';
+{{else}}
+import {DOMProps, DOMRef, StyleProps} from '@react-types/shared';
+{{/if}}
+import {filterDOMProps} from '@react-aria/utils';
 import React from 'react';
+{{#if (includes scopes "@react-types")}}
+import {Spectrum{{~componentName}}Props} from '@react-types/{{packageName}}';
+{{/if}}
+{{#if componentCSS}}
 import styles from '@adobe/spectrum-css-temp/components/{{componentCSS}}/vars.css';
+{{/if}}
+{{#if (includes scopes "@react-aria")}}
 import {use {{~componentName~}} } from '@react-aria/{{packageName}}';
+{{/if}}
+{{#if (includes scopes "@react-stately")}}
```

Comment:
I could perhaps by consuming the packageName and doing some more stuff with the if statements but perhaps for a followup pull, it might get a little hairy